src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,204 @@
+package parser;
+
+import commands.*;
+import duke.DukeException;
+
+import java.util.ArrayList;
+
+/**
+ * This class handles all string manipulation needed
+ * to read user inputs.
+ */
+public class Parser {
+
+    private String type = "";
+
+    private enum command {
+        bye("bye"),
+        list("list"),
+        mark("mark"),
+        unmark("unmark"),
+        deadline("deadline"),
+        event("event"),
+        todo("todo"),
+        delete("delete"),
+        find("find");
+
+        private final String command;
+
+        command(String command) {
+            this.command = command;
+        }
+
+        @Override
+        public String toString() {
+            return command;
+        }
+    }
+
+    /**
+     * Parser empty constructor.
+     */
+    public Parser(){
+    }
+
+    /**
+     * Parser constructor that takes in a String.
+     * @param type The type of task.
+     */
```

Comment:
Perhaps you may consider changing the phrasing to be "Constructs..." instead of the current "Constructor..."?

Reply:
Thx for the suggestion, will take note of it